src/Features/Core/Portable/Snippets/ConvertToLSPSnippetService.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.Snippets
+{
+    internal class ConvertToLSPSnippetService : ILanguageService
+    {
+        public string? GenerateLSPSnippet(TextChange textChange, List<(string, List<TextSpan>)> placeholders)
```

Comment:
For placeholders it might be better to represent them as a specific DTO. Might be a personal preference of mine though too 😄